e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/deal-cancellation-maker-unable-issue-facility.spec.js

```diff
@@ -0,0 +1,84 @@
+import portalPages from '../../../../../../../portal/cypress/e2e/pages';
+import MOCK_USERS from '../../../../../../../e2e-fixtures/portal-users.fixture';
+import getAinUnissuedDeals from '../../test-data/AIN-deal-unissued-facilities/dealReadyToSubmit';
+import getMinUnissuedDeals from '../../test-data/MIN-deal-unissued-facilities/dealReadyToSubmit';
+import { TFM_URL, PIM_USER_1 } from '../../../../../../../e2e-fixtures';
+
+import { yesterday, tomorrow } from '../../../../../../../e2e-fixtures/dateConstants';
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+context('Deal cancellation', () => {
+  const ainUnissuedDeals = Array(2).fill(getAinUnissuedDeals());
+  const minUnissuedDeals = Array(2).fill(getMinUnissuedDeals());
+  let deals = [];
+  const dealFacilities = [];
+
+  before(() => {
+    cy.insertManyDeals([...ainUnissuedDeals, ...minUnissuedDeals], BANK1_MAKER1).then((insertedDeals) => {
+      insertedDeals.forEach((insertedDeal) => {
+        cy.createFacilities(insertedDeal._id, insertedDeal.mockFacilities, BANK1_MAKER1).then((createdFacilities) => {
+          dealFacilities.push(...createdFacilities);
+        });
+      });
+      deals = [...insertedDeals];
+    });
+  });
+
+  beforeEach(() => {
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+    cy.login(BANK1_MAKER1);
+  });
+
+  after(() => {
+    cy.clearCookies();
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  describe('Deal cancellations on tfm with effective dates in the past and in the future', () => {
+    before(() => {
+      deals.forEach((deal, index) => {
+        const effectiveDate = index % 2 === 0 ? tomorrow.date : yesterday.date;
+        cy.makerSubmitDealForReview(deal);
+        cy.checkerSubmitDealToUkef(deal);
+
+        cy.clearCookie('dtfs-session');
+        cy.clearCookie('_csrf');
+        cy.getCookies().should('be.empty');
+
+        cy.forceVisit(TFM_URL);
+        cy.tfmLogin(PIM_USER_1);
+
+        cy.submitDealCancellation({ dealId: deal._id, effectiveDate });
+
+        cy.clearCookie('dtfs-session');
+        cy.clearCookie('_csrf');
+        cy.getCookies().should('be.empty');
+      });
+    });
+
+    it('AIN deal with unissued facilities is submitted to UKEF, user schedule cancellation in the future in TFM. Maker unable to issue facility on portal', () => {
+      portalPages.contract.visit(deals[0]);
+      portalPages.contract.bondTransactionsTable.row(dealFacilities[1]._id).issueFacilityLink().should('not.exist');
```

Comment:
Similar to my comment on https://github.com/UK-Export-Finance/dtfs2/pull/3938, I think `deals[0]` is confusing - it would be better to either use `deals.filter(({ tfm }) => tfm.cancellation.effectiveDate >= today )[0]` or, maybe easier to set these as constants at the top rather than using an array at all.

E.g. `const dealWithUnissuedFaciltitiesAndFutureCancellation` (quite a mouthful but more clear? Happy to think about better options too)

Reply:
I made some changes on this. I created the cy.getOneDeal function in order to have the deal status updated after the cancellation and then in `it`  i use `deals.find((deal) => deal.submissionType === 'Automatic Inclusion Notice' && deal.status === 'Cancelled');` to find the deal and `ainDealPast.facilities.find((facility) => facility.type === 'Bond')._id;` to find the facility.

Reply:
This feels a lot more robust - thank you :)